src/main/java/globalquake/ui/stationselect/StationSelectFrame.java

```diff
@@ -6,21 +6,61 @@
 import javax.swing.*;
 import java.awt.*;
 import java.awt.event.ActionEvent;
-import java.util.Timer;
-import java.util.TimerTask;
+import java.awt.event.ActionListener;
 
-public class StationSelectFrame extends JFrame {
+public class StationSelectFrame extends JFrame implements ActionListener {
 
     private final StationSelectPanel stationSelectPanel;
     private final DatabaseMonitorFrame databaseMonitorFrame;
     private JToggleButton selectButton;
     private JToggleButton deselectButton;
+    private JButton selectAll;
+    private JButton deselectAll;
+    private boolean stationRepaint = false;
     private DragMode dragMode = DragMode.NONE;
     private final JCheckBox chkBoxShowUnavailable;
 
     public StationSelectFrame(DatabaseMonitorFrame databaseMonitorFrame) {
         setLayout(new BorderLayout());
         this.databaseMonitorFrame = databaseMonitorFrame;
+        
+        JPanel togglePanel = new JPanel(new GridLayout(9,1));
+        JButton toggleButton = new JButton("<");
+        selectAll = new JButton(new SelectAllAction(databaseMonitorFrame.getManager()));
+        deselectAll = new JButton(new DeselectAllAction(databaseMonitorFrame.getManager()));
+        selectAll.addActionListener(this);
+        deselectAll.addActionListener(this);
+        JPanel filler1 = new JPanel();
+        JPanel filler2 = new JPanel();
+        JPanel filler3 = new JPanel();
+        JPanel filler4 = new JPanel();
+        JPanel filler5 = new JPanel();
+        JPanel filler6 = new JPanel();
+        JPanel filler7 = new JPanel();
+        JPanel filler8 = new JPanel();
+
+        filler1.setOpaque(false);
+        filler2.setOpaque(false);
+        filler3.setOpaque(false);
+        filler4.setOpaque(false);
+        filler5.setOpaque(false);
+        filler6.setOpaque(false);
+        filler7.setOpaque(false);
+        filler8.setOpaque(false);
+
+        togglePanel.add(filler1);
+        togglePanel.add(filler2);
+        togglePanel.add(filler3);
+        togglePanel.add(filler4);
+        togglePanel.add(toggleButton);
+        togglePanel.add(filler5);
+        togglePanel.add(filler6);
+        togglePanel.add(filler7);
+        togglePanel.add(filler8);
+        togglePanel.setOpaque(false);
+
+        toggleButton.setToolTipText("Toggle Toolbar");
+        toggleButton.setBackground(Color.GRAY);
 
         stationSelectPanel = new StationSelectPanel(this, databaseMonitorFrame.getManager());
```

Comment:
Here you need to override the `paint()` method, call `super.paint()` to paint the whole StationSelectPanel and after that call centerPanel.repaint(). This will solve all issues with flickering and unnecessary use of Timer.

Reply:
I didn't understand how to implement the `super.paint()` call. If I remember correctly this call requires a graphics method. I was not sure if you wanted me to implement one to repaint the frame. For now, I implemented your other suggestion and repainted the whole `centerPanel` using the timer which seems to work to get rid of the flickering.

Reply:
Hi! this is what I wanted to be done:
```
JPanel centerPanel = new JPanel(new GridBagLayout());
stationSelectPanel = new StationSelectPanel(this, databaseMonitorFrame.getManager()){
    @Override
    public void paint(Graphics gr) {
        super.paint(gr);
        centerPanel.repaint();
    }
};
```

I should've probably explained a little bit more about that super.paint() call...
This will make the centerPanel repaint every time after the main globe panel is painted, solving the issue with flickering after removing the timer workaround (it is flickering now much less, but still)